File: Array/xietao/6月3日—查到字符串起始位置/java/AnswerSolution.java
public class AnswerSolution {
    public int strStr(String haystack, String needle) {
        int L = needle.length(), n = haystack.length();
        if(L == 0){
            return 0;
        }
        int pn = 0;//这个是haystack字符串的指针
        while(pn < n - L + 1){
            while(pn < n - L +1 && haystack.charAt(pn) != needle.charAt(0)) ++pn;//找到起始的位置
            int currLen = 0;//匹配到相同字符串的长度
            int pL =0;//这是needle字符串的指针
            while(pL < L && pn < n && haystack.charAt(pn)==needle.charAt(pL)){
                ++pn;
                ++pL;
                ++currLen;
            }
            if(currLen == L){
                return pn - L;
            }
            pn = pn - currLen + 1;
        }
        return -1;
    }
}

File: Array/xietao/6月3日—查到字符串起始位置/C/answerSolution.c
//
// Created by 谢涛 on 2020/6/3.
//

#include <stdio.h>
#include <stdlib.h>

int strStr(char * haystack, char * needle);
int main() {
    char *haystack = "hwwwello";
    char *needle = "ll";
    int index = strStr(haystack,needle);
    printf("index=%d\n",index);
    return 0;
}

int strStr(char * haystack, char * needle){
    int L = strlen(needle);
    int n = strlen(haystack);
    if(L ==0) return 0;
    int pn = 0;
    while(pn < n - L +1){
        while(pn < n - L + 1 && haystack[pn] != needle[0]) ++pn;
        int currLen = 0, pL = 0;
        while(pL < L && pn < n && haystack[pn] == needle[pL]){
            ++pn;
            ++pL;
            ++currLen;
        }
        if(currLen == L){
            return pn - L;
        }
        pn = pn -currLen +1;
    }
    return -1;
}





